src/main/kotlin/pt/up/fe/ni/website/backend/controller/ProjectController.kt

```diff
@@ -24,20 +30,34 @@ class ProjectController(private val service: ProjectService) {
     @GetMapping("/{projectSlug}**")
     fun getProjectBySlug(@PathVariable projectSlug: String) = service.getProjectBySlug(projectSlug)
 
-    @PostMapping("/new")
-    fun createNewProject(@RequestBody dto: ProjectDto) = service.createProject(dto)
+    @PostMapping("/new", consumes = ["multipart/form-data"])
+    fun createProject(
+        @RequestPart project: ProjectDto,
+        @RequestParam
+        @ValidImage
+        image: MultipartFile
+    ): Project {
+        project.imageFile = image
+        return service.createProject(project)
+    }
 
     @DeleteMapping("/{id}")
     fun deleteProjectById(@PathVariable id: Long): Map<String, String> {
         service.deleteProjectById(id)
         return emptyMap()
     }
 
-    @PutMapping("/{id}")
+    @PutMapping("/{id}", consumes = ["multipart/form-data"])
     fun updateProjectById(
         @PathVariable id: Long,
-        @RequestBody dto: ProjectDto
-    ) = service.updateProjectById(id, dto)
+        @RequestPart project: ProjectDto,
+        @RequestParam
+        @ValidImage
+        image: MultipartFile?
+    ): Project {
+        project.imageFile = image
+        return service.updateProjectById(id, project)
+    }
```

Comment:
In this version, the way to update the timeline is to send a PUT with the whole project, the same way we do for the remaining fields. I think this makes the API simpler and is fine since the timeline doesn't hold a lot of data